### PR TITLE
Related items comparison

### PR DESCRIPTION
**Pull request by debrazhang09**
Add mouseover on icon star to show modal at the end of the related product list, click only image to change to a new page.
Add single unit test, add some css

### CODE REVIEW

File: client/src/components/relatedItems/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  constructor(props) {
+    super(props);
+    // this.el = document.createElement('div');
+    this.state = {
+      modal: document.createElement('div')
+    }
+  }
+  componentDidMount() {
+    const modalRoot = document.getElementsByClassName('modal-container')[0];
+    modalRoot.appendChild(this.state.modal);
```
Comment (nsfoster87) [inline client/src/components/relatedItems/Modal.jsx:16]: I think you can call `document.querySelector('.modal-container');` and that way you don't need to get an array and pull the first element from the array like with `getElementsByClassName`

File: client/src/components/relatedItems/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+
+export default class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    // this.el = document.createElement('div');
+    this.state = {
+      modal: document.createElement('div')
+    }
```
Comment (nsfoster87) [inline client/src/components/relatedItems/Modal.jsx:12]: I'm wondering if you need to keep the element itself inside this.state. It doesn't seem to update or change ever, so maybe using a property (`this.modal = document.createElement('div')`) would make more sense?

File: client/src/components/relatedItems/SingleCard.jsx [hunk 3]
```diff
@@ -131,27 +135,51 @@ export default class SingleCard extends React.Component {
...
+          }
+
+
+        </div>
+
+
+
+
+        <div style={{"marginLeft": "5px"}}>
+
```
Comment (nsfoster87) [inline client/src/components/relatedItems/SingleCard.jsx:176]: I always think it's best to keep styles inside the CSS sheet. Maybe put a className on the div and keep the style in the styles.css.

File: client/src/components/relatedItems/Modal.jsx (hunk 1 above)
Comment (debrazhang09) [reply client/src/components/relatedItems/Modal.jsx:16]: I changed to it and it works well, I'll change in the next pull request.Thanks, Nate.

**Review by nsfoster87 — APPROVED** [review]
My main thoughts are:
1. Does the modal element need to be held inside state?
2. Maybe use classNames on elements and keep the styles inside syles.css

File: client/src/components/relatedItems/Modal.jsx (hunk 2 above)
Comment (debrazhang09) [reply client/src/components/relatedItems/Modal.jsx:12]: Yes, it worked. I thought if something changed in componentDidMount, it should relate to state change. `this.setState` works on it

File: client/src/components/relatedItems/SingleCard.jsx (hunk 3 above)
Comment (debrazhang09) [reply client/src/components/relatedItems/SingleCard.jsx:176]: OK. I think it's the best practice of css. I'll change it in next pull request

**Pull request merged**